extras/buildah/build.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+#
+# Build heketi containers using the 'buildah' tool either from source
+# or from existing compiled binaries.
+#
+
+import argparse
+import os
+import shutil
+import subprocess
+import tempfile
+
+USE_SUDO = True
+
+pjoin = os.path.join
+
+
+def buildah(*args, redirect=False):
+    """A lightweight wrapper around buildah command."""
+    cmd = []
+    if USE_SUDO and os.getuid() != 0:
+        cmd.append("sudo")
+    cmd.append("buildah")
+    cmd.extend(args)
+    if redirect:
+        return subprocess.check_output(cmd).strip().decode('utf8')
+    else:
+        subprocess.check_call(cmd)
+
+
+def buildah_from(src):
+    return buildah('from', src, redirect=True)
+
+
+def build_binaries(outdir):
+    pkgs = 'glide golang git make mercurial'
+    hdir = '/build/src/github.com/heketi/heketi'
+    heketi_branch = 'master'
+    heketi_url = 'https://github.com/heketi/heketi.git'
+
+    bc = buildah_from('fedora:27')
+    buildah('run', bc, 'dnf', '-y', 'install', *pkgs.split())
+    buildah('run', bc, 'mkdir', '-p', os.path.dirname(hdir))
+    buildah('run', bc, 'git', '-C', os.path.dirname(hdir),
+            'clone', '-b', heketi_branch, heketi_url)
+    buildah('config',
+            '--workingdir', hdir,
+            '--env', 'GOPATH=/build',
+            bc)
+    buildah('run', bc, 'make')
+
+    mnt = buildah('mount', bc, redirect=True)
+    heketi_dir = pjoin(mnt, hdir.strip('/'))
+    heketi_bin = pjoin(heketi_dir, 'heketi')
+    heketi_cli_bin = pjoin(heketi_dir, 'client/cli/go/heketi-cli')
+    shutil.copy(heketi_bin, outdir)
+    shutil.copy(heketi_cli_bin, outdir)
+    buildah('umount', bc)
+
+    buildah('rm', bc)
+
+
+def build_container(sources):
+    hc = buildah_from('fedora:27')
```

Comment:
This could probably be `fedora:latest`?

Reply:
There was an issue using fedora 28 IIRC. What I should do is move that to a top-level variable and make it possible to swap the base image easier for testing.